app/src/sharedTest/java/org/oppia/app/topic/TopicFragmentTest.kt

```diff
@@ -0,0 +1,350 @@
+  fun testTopicFragment_clickOnTrainTab_showsTrainTabWithContentMatched() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText(R.string.train),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withText("Master These Skills")).check(matches(isDisplayed()))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicFragment_clickOnReviewTab_showsReviewTabSelected() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText(R.string.review),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withId(R.id.topic_tabs_container)).check(matches(matchCurrentTabTitle(getResourceString(R.string.review))))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicFragment_clickOnReviewTab_showsReviewTabWithContentMatched() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText(R.string.review),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(
+      allOf(
+        atPosition(R.id.review_skill_recycler_view, 0).also { withText("An important skill") },
+        isDescendantOfA(withId(R.id.review_skill_recycler_view))
+      ).also { withId(R.id.skill_name) })
+  }
+
+  @Test
+  @UiThread
+  fun testTopicFragment_clickOnOverviewTab_showsOverviewTabSelected() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText(R.string.review),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(
+      allOf(
+        withText(R.string.overview),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withId(R.id.topic_tabs_container)).check(matches(matchCurrentTabTitle(getResourceString(R.string.overview))))
+  }
+
+  @Test
+  @UiThread
+  fun testTopicFragment_clickOnOverviewTab_showsOverviewTabWithContentMatched() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText(R.string.review),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(
+      allOf(
+        withText(R.string.overview),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withId(R.id.topic_name_text_view)).check(
+      matches(
+        withText(
+          Matchers.containsString(topicName)
+        )
+      )
+    )
+  }
+
+  @Test
+  @UiThread
+  fun testTopicFragment_clickOnPlayTab_configurationChange_showsSameTabAndItsData() {
+    activityTestRule.launchActivity(null)
+    onView(
+      allOf(
+        withText(R.string.play),
+        isDescendantOfA(withId(R.id.topic_tabs_container))
+      )
+    ).perform(click())
+    onView(withText("First Story")).check(matches(isDisplayed()))
+    activityTestRule.activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
```

Comment:
Doesn't this need to happen in an ``onActivity()`` callback? As I recall from documentation, requestedOrientation should only be called on the main thread. Ditto elsewhere.

Reply:
Is this the solution you are suggesting ,Please take a look on this link [https://github.com/oppia/oppia-android/pull/289#discussion_r342008956 ](url). 
It seems not working.
Please do suggest the link if we have any google documentation for this fix.

Reply:
Resolved